### PR TITLE
Feature flags

### PR DESCRIPTION
**Pull request by mjudeikis**
### Which issue this PR addresses:

Adds feature flags into main controllers in the operator

### What this PR does / why we need it:

We need this so we can control better behavior of the individual components in the operator when issues are rasaised
SystemReserved and routeFix are good examples when we needed this

### Test plan for issue:

Unit tests

### Is there any documentation that needs to be updated for this PR?

No

### CODE REVIEW

**Review by gvanderpotte — COMMENTED** [review]
Looks good to me. One small nit: Should not the unit tests be updated to check that the various controllers are deactivated correctly if the featureflag is set to false ?

**Comment by mjudeikis** [comment]
> Looks good to me. One small nit: Should not the unit tests be updated to check that the various controllers are deactivated correctly if the featureflag is set to false ?

easier to say than to do :/ Need to refactor tests. Will see if I have time to get to this

File: pkg/operator/controllers/workaround/workaround_controller.go [hunk 1]
```diff
@@ -51,6 +52,16 @@ func NewReconciler(log *logrus.Entry, kubernetescli kubernetes.Interface, config
 
 // Reconcile makes sure that the workarounds are applied or removed as per the OpenShift version.
 func (r *WorkaroundReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	instance, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		r.log.Error(err)
```
Comment (m1kola) [inline pkg/operator/controllers/workaround/workaround_controller.go:57]: Why we log here, but not in other places? I think controller-runtime will log errors for us, so this logging is not necessary.

**Review by m1kola — APPROVED** [review]

**Pull request merged**